fe/fe-core/src/main/java/com/starrocks/http/rest/FeatureAction.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/http/rest/HealthAction.java
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.http.rest;
+
+import com.starrocks.common.Version;
+import com.starrocks.http.ActionController;
+import com.starrocks.http.BaseRequest;
+import com.starrocks.http.BaseResponse;
+import com.starrocks.http.IllegalArgException;
+import io.netty.handler.codec.http.HttpMethod;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FeatureAction extends RestBaseAction {
+    public FeatureAction(ActionController controller) {
+        super(controller);
+    }
+
+    public static void registerAction(ActionController controller)
+            throws IllegalArgException {
+        controller.registerHandler(HttpMethod.GET, "/api/feature", new FeatureAction(controller));
+    }
+
+    @Override
+    public void execute(BaseRequest request, BaseResponse response) {
+        response.setContentType("application/json");
+
+        RestResult result = new RestResult();
+        result.addResultEntry("version", Version.STARROCKS_VERSION);
+        result.addResultEntry("features", Feature.getFeatures());
+        sendResult(request, response, result);
+    }
+}
+
+class Feature {
```

Comment:
rename to `ProductFeature` and put into separate java file.

Reply:
```
{
  "features": [
    {
      "name": "Feature Name",
      "version": "1.0.0",
      "description": "Feature Description",
      "link": "https://github.com/starrocksdb/starrocks/issues/new"
    }
  ],
  "version": "feature/add-api-feature-interface",
  "status": "OK"
}
```